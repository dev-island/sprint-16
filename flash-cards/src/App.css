.card {
  padding: 2em;
}


.center {
  display: flex;
  justify-content: center;
  align-items: center;
}

.space {
  margin: 0 1em;
}

.flex-end {
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

label {
  font-size: 1.2em;
  font-weight: 500;
  display: block;
  color: #646cff;
}

input {
  border: 1px solid #ccc;
  background-color: #ccc;
  border-radius: 4px;
  padding: 0.5em;
  font-size: 1em;
  font-family: inherit;
  width: 100%;
  box-sizing: border-box;
}