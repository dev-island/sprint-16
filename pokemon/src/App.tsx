import { useState } from 'react';
import PokemonList from "./views/PokemonList";
import Header from "./components/Header";
import Pokemon from './views/Pokemon';
import { Box } from "@chakra-ui/react";

function App() {
  const [currentPokemonId, setCurrentPokemonId] = useState<number | null>(null);
  return (
    <Box>
      <Header goBack={() => setCurrentPokemonId(null)}/>
      {currentPokemonId ? (
        <Pokemon id={currentPokemonId} />
      ) : (
        <PokemonList
          viewPokemon={(id: number) => setCurrentPokemonId(id)}
        />
      )}

    </Box>
  );
}

export default App;
