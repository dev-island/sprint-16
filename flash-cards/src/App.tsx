import "./App.css";
import NavBar from "./components/NavBar";
import styled from "styled-components";
import useAppContext from "./hooks/useAppContext";
import Unknown from "./views/404";
import Home from "./views/HomeView";
import Stats from "./views/StatsView";
import Study from "./views/StudyView";
import ListCards from "./views/ListCardsView";
import Settings from "./views/SettingsView";
import { AppView } from "./context/application/types";
import { FC } from "react";
import CardForm from "./components/CardForm";
import styles from './app.module.css';

const VIEW_MAP: Record<AppView, FC> = {
  [AppView.HOME]: Home,
  [AppView.STUDY]: Study,
  [AppView.STATS]: Stats,
  [AppView.LIST]: ListCards,
  [AppView.SETTINGS]: Settings,
};

const screenSizes = {
  mobile: '(min-width: 768px)',
  tablet: '(min-width: 1024px)',
  desktop: '(min-width: 1280px)',
}

function App() {
  const { currentView } = useAppContext();

  const View = VIEW_MAP[currentView] || Unknown;
  return (
    <Container>
      <NavBar />
      <CardForm />
      <Content fullScreen>
        <View />
      </Content>
    </Container>
  );
}

export default App;

const Container = styled.div`

  background: green;

  @media ${screenSizes.mobile} {
    background: red;
  }
`

const Content = styled.div<{fullScreen?: boolean}>`
  padding: 2rem;
  flex: 1;

  ${({fullScreen}) => fullScreen && `
    height: 100%;
  `}
`;
