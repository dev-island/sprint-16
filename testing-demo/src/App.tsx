import { useState } from 'react'
import './App.css'
import Search from './components/Search';

function App() {
  const [search, setSearch] = useState('');

  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
    setSearch(event.target.value);
  }

  return (
    <div>
      <Search value={search} onChange={handleChange}>
        Search:
      </Search>

      <p data-testid="search-text">Searches for {search ? search : "..."}</p>
    </div>
  );
}

export default App
