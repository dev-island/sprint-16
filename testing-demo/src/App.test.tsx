import { describe, it, expect } from "vitest";
import App from "./App";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";


describe("something truthy and falsy", () => {
  it("true to be true", () => {
    expect(true).toBe(true);
  });

  it("false to be false", () => {
    expect(false).toBe(false);
  });
});

describe("App", () => {
  it("renders App component", () => {
    render(<App />);
  })

  it("matches the snapshot App component", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  })

  it('should render the search bar', () => {
    render(<App />);
    expect(screen.queryByText(/Searches for JavaScript/)).toBeNull();

    fireEvent.change(screen.getByRole("textbox"), {
      target: { value: "JavaScript" },
    });
    // Can use "waitFor" to wait for the DOM to update
    // Or 
    waitFor(() =>
      expect(screen.getByText(/Searches for JavaScript/)).toBeInTheDocument()
    );
  });
});