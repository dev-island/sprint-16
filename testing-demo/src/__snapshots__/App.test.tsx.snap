// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > matches the snapshot App component 1`] = `
<div>
  <div>
    <div>
      <label
        for="search"
      >
        Search:
      </label>
      <input
        id="search"
        type="text"
        value=""
      />
    </div>
    <p>
      Searches for 
      ...
    </p>
  </div>
</div>
`;
